pkg/manager/member/tidb_member_manager.go

```diff
@@ -198,7 +198,7 @@ func (tmm *tidbMemberManager) syncTiDBStatefulSetForTidbCluster(tc *v1alpha1.Tid
 		}
 	}
 
-	if tmm.autoFailover {
+	if tmm.autoFailover && tc.Spec.TiDB.MaxFailoverCount != nil {
```

Comment:
what's the default value?

Reply:
I seem that we don't have a default value, so the user must specify this field to enable the auto-failover feature? @aylei @weekface what do you think?

Reply:
Yes, it is ok to me to set the default value to ensure the `MaxFailoverCount` is not nil`.  But I still think it is necessary to check whether the `MaxFailoverCount` is nil here.

Reply:
The `tc.Spec.TiDB.MaxFailoverCount` will never be nil after adding the defaulting function, right?